src/absil/ilreflect.fs

```diff
@@ -141,58 +144,64 @@ type System.Reflection.Emit.MethodBuilder with
         methB
 
-    member typB.DefineGenericParametersAndLog(gps)        = 
+    member typB.DefineGenericParametersAndLog(gps) = 
         if logRefEmitCalls then printfn "typeBuilder%d.DefineGenericParameters(%A)" (abs <| hash typB) gps
         typB.DefineGenericParameters(gps)
 
-    member typB.DefineConstructorAndLog(attrs,cconv,parms)        = 
+    member typB.DefineConstructorAndLog(attrs,cconv,parms) = 
         let consB = typB.DefineConstructor(attrs,cconv,parms)
         if logRefEmitCalls then printfn "let constructorBuilder%d = typeBuilder%d.DefineConstructor(enum %d,%A,%A)" (abs <| hash consB) (abs <| hash typB) (LanguagePrimitives.EnumToValue attrs) cconv parms
         consB
 
-    member typB.DefineFieldAndLog(nm,ty:System.Type,attrs)        = 
+    member typB.DefineFieldAndLog(nm,ty:System.Type,attrs) = 
         if logRefEmitCalls then printfn "typeBuilder%d.DefineField(\"%s\",typeof<%s>,enum %d)" (abs <| hash typB) nm ty.FullName (LanguagePrimitives.EnumToValue attrs)
         typB.DefineField(nm,ty,attrs)
 
-    member typB.DefinePropertyAndLog(nm,attrs,ty,args)        = 
+    member typB.DefinePropertyAndLog(nm,attrs,ty,args) = 
         if logRefEmitCalls then printfn "typeBuilder%d.DefineProperty(\"%A\",enum %d,%A,%A)" (abs <| hash typB) nm (LanguagePrimitives.EnumToValue attrs) ty args
         typB.DefineProperty(nm,attrs,ty,args)
 
-    member typB.DefineEventAndLog(nm,attrs,ty)        = 
+    member typB.DefineEventAndLog(nm,attrs,ty) = 
         if logRefEmitCalls then printfn "typeBuilder%d.DefineEvent(\"%A\",enum %d,%A)" (abs <| hash typB) nm (LanguagePrimitives.EnumToValue attrs) ty
         typB.DefineEvent(nm,attrs,ty)
 
-    member typB.SetParentAndLog(ty:System.Type)        = 
+    member typB.SetParentAndLog(ty:System.Type) = 
         if logRefEmitCalls then printfn "typeBuilder%d.SetParent(typeof<%s>)" (abs <| hash typB) ty.FullName
         typB.SetParent(ty)
 
-    member typB.AddInterfaceImplementationAndLog(ty)        = 
+    member typB.AddInterfaceImplementationAndLog(ty) = 
         if logRefEmitCalls then printfn "typeBuilder%d.AddInterfaceImplementation(%A)" (abs <| hash typB) ty
         typB.AddInterfaceImplementation(ty)
 
-    member typB.InvokeMemberAndLog(nm,flags,args)        = 
-        if logRefEmitCalls then printfn "typeBuilder%d.InvokeMember(\"%s\",enum %d,null,null,%A,Globalization.CultureInfo.InvariantCulture)" (abs <| hash typB) nm (LanguagePrimitives.EnumToValue flags) args     
-        typB.InvokeMember(nm,flags,null,null,args,Globalization.CultureInfo.InvariantCulture)
+    member typB.InvokeMemberAndLog(nm,_flags,args) = 
+#if FX_RESHAPED_REFEMIT
+        let t = typB.CreateTypeAndLog()
+        let m =  if t <> null then t.GetMethod(nm, (args |> Seq.map(fun x -> x.GetType()) |> Seq.toArray)) else null
```

Comment:
The ``if .. null...`` isn't helpful since a ``null`` will just cause an NRE on the next line anyway